elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -304,21 +305,34 @@ private String extractJoin(Path.PathElement pathElement) {
         Class<?> entityClass = pathElement.getType();
         String entityAlias = FilterPredicate.getTypeAlias(entityClass);
 
-        Table tableAnnotation = dictionary.getAnnotation(relationshipClass, Table.class);
-
-        String relationshipTableName = (tableAnnotation == null)
-                ? dictionary.getJsonAliasFor(relationshipClass)
-                : tableAnnotation.name();
 
         String relationshipIdField = getColumnName(relationshipClass, dictionary.getIdFieldName(relationshipClass));
-
-        return String.format("LEFT JOIN %s AS %s ON %s.%s = %s.%s",
-                relationshipTableName,
-                relationshipAlias,
-                entityAlias,
-                getColumnName(entityClass, relationshipName),
-                relationshipAlias,
-                relationshipIdField);
+        String relationshipColumnName = getColumnName(entityClass, relationshipName);
+        Subselect subselectAnnotation = dictionary.getAnnotation(relationshipClass, Subselect.class);
```

Comment:
I think it will be more maintainable long term if all of the dictionary lookups can be moved to the Schema classes.   When we build the SQL schema, each column should know how to join to it.

Reply:
I move the code into `SQLSchema`, but I don't think we need to cache the join clause, they should just be used once, right?